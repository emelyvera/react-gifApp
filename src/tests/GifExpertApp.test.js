import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas del componente <GifExpertApp />', () => {
  
  test('should funcionar todo correctamente', () => {

    const wrapper = shallow( <GifExpertApp /> );
    expect( wrapper ).toMatchSnapshot();

  });

  test('should de mostar una lista de ctaegorias', () => {
    const category = ['One Punch', 'Batman'];
    const wrapper = shallow( <GifExpertApp defaultCategories= {category} /> );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe( category.length );
    
  });
  
    


});
