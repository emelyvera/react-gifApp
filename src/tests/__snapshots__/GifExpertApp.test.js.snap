// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifExpertApp /> should de mostar una lista de ctaegorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Batman"
      key="Batman"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas del componente <GifExpertApp /> should funcionar todo correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
